lib/fittings/swagger_serdes.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+var debug = require('debug')('swagger:serialize');
+var path = require('path');
+var _ = require('lodash');
+
+module.exports = function create(fittingDef, bagpipes) {
+    var swaggerNodeRunner = bagpipes.config.swaggerNodeRunner;
+    var appRoot = swaggerNodeRunner.config.swagger.appRoot;
+    var serdesDirs = fittingDef.serdesDirs.map(function(dir) {
+        return path.resolve(appRoot, dir);
+    });
+    var serdesFunctionCache = {};
+
+    return function swaggerSerdes(context, next) {
+        var serdesNameKey = fittingDef.serdesNameKey;
+        var operation = context.request.swagger.operation;
+        var serdesName;
+        var serdes;
+
+        try {
+            serdesName = operation[serdesNameKey] || operation.pathObject[serdesNameKey];
+            if (!serdesName) { return next(); }
+        } catch (err) {
+            // serdes is optional, hand off to the next fitting if serdesName is undefined.
+            return next();
+        }
+
+        if (serdesName in serdesFunctionCache) {
+            serdes = serdesFunctionCache[serdesName];
+        } else {
+            for(var i = 0; i < serdesDirs.length; i++) {
```

Comment:
@jlongever Not in this case because of the test for i === length in the catch block.  This code was lifted from the swagger_router fitting that comes with swagger-node-runner.